docs-src/java/how-to-use-a-custom-payload-codec-in-java.md

```diff
@@ -0,0 +1,111 @@
+---
+id: how-to-use-a-custom-payload-codec-in-java
+title: How to use a custom Payload Codec in Java
+sidebar_label: Custom Payload Codec
+description: Create a custom implementation of `PayloadCodec` and use it in `CodecDataConverter` to set a custom Data Converter.
+tags:
+  - Java
+  - developer-guide
+---
+
+**Create a custom Payload Codec**
+
+Create a custom implementation of [`PayloadCodec`](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/payload/codec/PayloadCodec.html) and use it in [`CodecDataConverter`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/common/converter/CodecDataConverter.html)) to set a custom Data Converter.
+
+The Payload Codec does byte-to-byte conversion and must be set with a Data Converter to do the conversion to bytes and values.
+
+Define custom encryption/compression logic in your `encode` method, and decryption/decompression logic in your `decode` method.
+
+The following example from the [Java encryption sample](https://github.com/temporalio/samples-java/blob/main/src/main/java/io/temporal/samples/encryptedpayloads/CryptCodec.java) shows how to implement encryption and decryption logic on your payloads in your `encode` and `decode` methods.
+
+```java
+class YourCustomPayloadCodec implements PayloadCodec {
+  //...
+
+  // See the linked sample for details on the methods called here.
+  @NotNull
+  @Override
+  public List<Payload> encode(@NotNull List<Payload> payloads) {
+    return payloads.stream().map(this::encodePayload).collect(Collectors.toList());
+  }
+
+  @NotNull
+  @Override
+  public List<Payload> decode(@NotNull List<Payload> payloads) {
+    return payloads.stream().map(this::decodePayload).collect(Collectors.toList());
+  }
+
+  private Payload encodePayload(Payload payload) {
+    String keyId = getKeyId();
```

Comment:
If you carry usages of these methods but don't want to carry an implementation, you should as least save the method definition with an empty body and add a comment that "this should do this and that and a specific implementation is not important for this sample."

Reply:
Yep; adding comments inline for these